lib/superstore/types/time_type.rb

```diff
@@ -30,6 +30,10 @@ def decode(str)
       rescue ArgumentError
         Time.parse(str).in_time_zone rescue nil
       end
+
+      def typecast(value)
+        value.to_time.in_time_zone rescue nil
+      end
```

Comment:
It was possible to instantiate time types with non-time values (like integers)

Reply:
I would focus on making it match up with ActiveModel/ActiveRecord behavior.

One of my goals is to actually use ActiveModel and remove all of this code under superstore/types, so the more resemblance to active_model/types, the better